src/Pangraph/GraphML/Parser.hs

```diff
@@ -1,28 +1,21 @@
-module Pangraph.GraphML.Parser(
-parseGraphToAlga,
-parseTemplateToPangraph,
-template,
-PT.Template
+module Pangraph.GraphML.Parser (
+parse,
+unsafeParse
 ) where
 
-import qualified Pangraph as P
-import qualified Text.XML.Hexml as H
-import qualified Data.ByteString as BS
-import qualified Pangraph.XMLTemplate as PT
-import qualified Algebra.Graph.HigherKinded.Class as H
+import qualified Data.ByteString            as BS
+import qualified Pangraph                   as P
+import qualified Pangraph.Internal.Hexml    as H
+import qualified Pangraph.XMLTemplate       as PT
 
+parse :: BS.ByteString -> Maybe P.Pangraph
+parse file =
+  case H.hexmlParse file of
+    Nothing -> Nothing
+    Just hexml -> PT.hexmlToPangraph PT.graphMLTemplate hexml
 
-parseGraphToAlga:: (H.Graph g) => PT.Template -> BS.ByteString -> Either BS.ByteString (g P.Node)
-parseGraphToAlga t file =
-  case H.parse file of
-    Left x -> Left x
-    Right x -> Right $ PT.parseTemplateToAlga t x
-
-parseTemplateToPangraph :: PT.Template -> BS.ByteString -> Either BS.ByteString P.Pangraph
-parseTemplateToPangraph t file =
-  case H.parse file of
-    Left x -> Left x
-    Right x -> Right $ PT.parseTemplateToPangraph t x
-
-template:: [PT.Template]
-template = PT.graphMLTemplate
+unsafeParse :: BS.ByteString -> P.Pangraph
+unsafeParse file =
```

Comment:
You almost never need to examine `Maybe` in a case expression. Here you can do:

`unsafeParse = fromMaybe (error $ "Parse failed")`